src/trees/NodeAllocator.cpp

```diff
@@ -339,10 +339,11 @@ template <int D> int NodeAllocator<D>::findNextOccupied(int sIdx) const {
     assert(sIdx < this->stackStatus.size());
     bool posIsAvailable = (this->stackStatus[sIdx] == 0);
     bool endOfStack = (sIdx >= this->topStack);
-    while (posIsAvailable and not(endOfStack)) {
-        sIdx++;
-        posIsAvailable = (this->stackStatus[sIdx] == 0);
+    while (1) {
         endOfStack = (sIdx >= this->topStack);
+        if(endOfStack) break;
+        posIsAvailable = (this->stackStatus[sIdx] == 0);
+        if(!posIsAvailable) break;
```

Comment:
I guess `sIdx` still needs to be incremented? Looks like an infinite loop

Reply:
Yeah sorry, I didn't port the patch properly. Fix coming

Reply:
Now I copied the real patch that I used in the Fedora package that's already run 46 iterations successfully when it used to crash after 2-6